src/transformers/modeling_t5.py

```diff
@@ -837,16 +968,29 @@ def forward(
 
         hidden_states = encoder_outputs[0]
 
+        # If decoding with past key value states, only the last tokens
+        # should be given as an input
+        if decoder_past_key_value_states is not None and self.decoder.output_past is True:
+            if decoder_input_ids is not None:
+                decoder_input_ids = decoder_input_ids[:, -1:]
+            if decoder_inputs_embeds is not None:
+                decoder_inputs_embeds = decoder_inputs_embeds[:, -1:]
```

Comment:
That's a rather import design choice I made and which is different to GPT2. 
Even when the user is using key value caching, it is quite natural to still input all the input_ids (from `1` to `sequence_length` instead of just the last one as is done in GPT2). So using these lines of code it works both on inputting only the last `input_ids` and all of them.

Reply:
This seems like a good idea to me, but that would definitely need to be referenced somewhere.

I think eventually we would like to do the same to GPT-2, in order to have a consistent API.

Reply:
Noted down to do a PR for GPT2 to implement that as well :-)

Reply:
And updated the docstring that should clearly describe the behavior now.